public/lib/helpers/timeline_helper.js

```diff
@@ -50,15 +61,66 @@ define(function (require) {
      */
     TimelineHelper.prototype.pluckLabel = function (hit, params, notify) {
       let field;
-      if (params.labelFieldSequence) { // in kibi, we have the path property of a field
-        field = kibiUtils.getValuesAtPath(hit._source, params.labelFieldSequence);
+      let value;
+
+      // in kibi, we have the path property of a field
+      if (params.labelFieldSequence || this.isMultiField(hit, params.labelField)) {
+        field = params.labelFieldSequence;
+        value = kibiUtils.getValuesAtPath(hit._source, field);
+        // get value from hit.fields in case of multi-fields
+        if (!value || !value.length) {
+          field = !field || !field.length ? params.labelField : field.join('.');
+          // '' if the field value is null
+          value = !hit.fields || !hit.fields[field] || hit.fields[field][0] === '' ? undefined : hit.fields[field];
+        }
       } else {
-        field = _.get(hit._source, params.labelField);
+        if (params.labelField) {
+          field = params.labelField;
+          value = _.get(hit._source, field);
+          if (!value) {
+            value = !hit.fields || !hit.fields[field] || hit.fields[field] === '' ? undefined : hit.fields[field];
+          }
+        }
       }
-      if (field && (!_.isArray(field) || field.length)) {
-        return field;
+
+      return value && (!_.isArray(value) || value.length) ? value : 'N/A';
+    };
+
+    /**
+     * pluckDate returns date field value/raw value
+     *
+     * @param hit the document of the event
+     * @param field to represent params.startField or params.endField
+     * @param fieldSequence to represent params.startFieldSequence or params.endFieldSequence
+     * @returns object with two properties: value and raw value
+     */
+    TimelineHelper.prototype.pluckDate = function (hit, field, fieldSequence) {
```

Comment:
Do you mean the `pluckDate` method? I just refactored [this duplication](https://github.com/sirensolutions/kibi_timeline_vis/pull/117/files/1af5594fd97fe867104d0e0dc4a549bade927789#diff-7419febb00f92050ffc6047cee9067a6L211) in the code. The code from line 211 till line 219 is a duplicate of the code above. 
As far as I understand `kibiUtils.getValuesAtPath()` is used to get value in case of Kibi. And `_.get(hit._source, params.endField)` - in case of Kibana.
Plus, I get multifields inside the `pluckDate` method.